modules/contxtfulRtdProvider.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Contxtful Technologies Inc
+ * This RTD module provides receptivity feature that can be accessed using the
+ * getReceptivity() function. The value returned by this function enriches the ad-units
+ * that are passed within the `getTargetingData` functions and GAM.
+ *
+ */
+
+import { submodule } from '../src/hook.js';
+import {
+  logInfo,
+  logError,
+  isStr,
+  isEmptyStr,
+  buildUrl,
+} from '../src/utils.js';
+import { loadExternalScript } from '../src/adloader.js';
+
+const MODULE_NAME = 'contxtful';
+const MODULE = `${MODULE_NAME}RtdProvider`;
+
+const CONTXTFUL_RECEPTIVITY_DOMAIN = 'api.receptivity.io';
+
+let initialReceptivity = null;
+let contxtfulModule = null;
+
+/**
+ * Init function used to start sub module
+ * @param { { params: { version: String, customer: String } } } config
+ * @return { Boolean }
+ */
+function init(config) {
+  logInfo(MODULE, 'init', config);
+  initialReceptivity = null;
+  contxtfulModule = null;
+
+  try {
+    const {version, customer} = extractParameters(config);
+    initCustomer(version, customer);
+    return true;
+  } catch (error) {
+    logError(MODULE, error);
+    return false;
+  }
+}
+
+/**
+ * Extract required configuration for the sub module.
+ * validate that all required configuration are present and are valid.
+ * Throws an error if any config is missing of invalid.
+ * @param { { params: { version: String, customer: String } } } config
+ * @return { { version: String, customer: String } }
+ * @throws params.{name} should be a non-empty string
+ */
+function extractParameters(config) {
+  const version = config?.params?.version;
+  if (!isStr(version) || isEmptyStr(version)) {
+    throw Error(`${MODULE}: params.version should be a non-empty string`);
+  }
+
+  const customer = config?.params?.customer;
+  if (!isStr(customer) || isEmptyStr(customer)) {
+    throw Error(`${MODULE}: params.customer should be a non-empty string`);
+  }
+
+  return {version, customer};
+}
+
+/**
+ * Initialize sub module for a customer.
+ * This will load the external resources for the sub module.
+ * @param { String } version
+ * @param { String } customer
+ */
+function initCustomer(version, customer) {
+  const CONNECTOR_URL = buildUrl({
```

Comment:
please prefer connector_url on config if supplied so publishers can host and lock your script

Reply:
Thank you for your insightful review @patmmccann!
We use our infrastructure intentionally to ensure simplicity and seamless updates for our customers.
Considering our product’s SaaS nature, we prioritize stability and ease of use. We’d prefer to avoid encouraging unsupported self-hosting which could complicate the setup and update process for publishers.
Does that sound reasonable?
Let me know, thanks a lot for your time!

Reply:
Hello @patmmccann !
Thank you again for your time -- I wanted to double check with you if our use case seems reasonable to you? I can provide more details of course.
Best.

Reply:
Hello @patmmccann ,
I addressed it in my latest commit; so publishers may now self-host and lock the version if they prefer.
Thanks for the reviewing!
Best.